clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreBufferService.java

```diff
@@ -516,6 +544,58 @@ public boolean equals(Object o) {
     }
   }
 
+  static class CommandQueueNode extends QueueNode {
+    enum CommandType {
+      // only supports SYNC_OFFSET command today.
+      SYNC_OFFSET
+    }
+
+    private final CompletableFuture<Void> cmdExecutedFuture = new CompletableFuture<>();
+
+    private final CommandType commandType;
+
+    private boolean valid;
+
+    public CommandQueueNode(
+        CommandType commandType,
+        PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> consumerRecord,
+        StoreIngestionTask ingestionTask) {
+      super(consumerRecord, ingestionTask, StringUtils.EMPTY, 0);
+      this.commandType = commandType;
+      this.valid = true;
+    }
+
+    public CompletableFuture<Void> getCmdExecutedFuture() {
+      return cmdExecutedFuture;
+    }
+
+    public CommandType getCommandType() {
+      return commandType;
+    }
+
+    public synchronized void invalidate() {
+      valid = false;
+    }
+
+    // This method is used to execute the command synchronously. Once it is in execution, it cannot be invalidated.
+    public synchronized void executeSync(Runnable runnable) {
+      if (valid) {
+        runnable.run();
+      }
+      cmdExecutedFuture.complete(null);
+    }
+
+    @Override
+    public int hashCode() {
+      return super.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      return super.equals(o);
+    }
```

Comment:
Since these methods are not doing anything special, we can remove them, right?

Reply:
I gave a try to remove it and it turns out that SpotBug had an opinion:

> This class extends a class that defines an equals method and adds fields, but doesn't define an equals method itself. Thus, equality on instances of this class will ignore the identity of the subclass and the added fields. Be sure this is what is intended, and that you don't need to override the equals method. Even if you don't need to override the equals method, consider overriding it anyway to document the fact that the equals method for the subclass just return the result of invoking super.equals(o).

Reply:
Sure, we can keep them.